device/pcie/pci_device.hpp

```diff
@@ -34,84 +34,144 @@ struct PciDeviceInfo
 
-    PCIDevice(int device_id, int logical_device_id);
+    /**
+     * PCI device constructor.
+     *
+     * Opens the character device file descriptor, reads device information from
+     * sysfs, and maps device memory region(s) into the process address space.
+     * 
+     * @param device_number         N in /dev/tenstorrent/N
+     * @param logical_device_id     unique identifier for this device in the network topology
+     */
+    PCIDevice(int device_number, int logical_device_id = 0);
+
+    /**
+     * PCIDevice destructor.
+     * Unmaps device memory and closes chardev file descriptor.
+     */
     ~PCIDevice();
+
     PCIDevice(const PCIDevice&) = delete; // copy
     void operator=(const PCIDevice&) = delete; // copy assignment
 
+    /**
+     * @return PCI device info
+     */
     const PciDeviceInfo get_device_info() const { return info; }
 
+    /**
+     * @return which NUMA node this device is associated with, or -1 if non-NUMA
+     */
+    int get_numa_node() const { return numa_node; }
+
+    /**
+     * @return underlying file descriptor
+     * TODO: this is an abstraction violation to be removed when this class
+     * assumes control over hugepage/DMA mapping code.
+     */
+    int get_fd() const { return fd; }
+
+    /**
+     * @return N in /dev/tenstorrent/N
+     * TODO: target for removal; upper layers should not care about this.
+     */
+    int get_device_num() const { return device_num; }
+
+    /**
+     * @return unique integer for each device in entire network topology
+     * TODO: target for removal; upper layers shouldn't to pass this in here. It
+     * is unused by this class.
+     */
+    int get_logical_id() const { return logical_id; }
+
+    /**
+     * @return PCI device id
+     */
+    int get_pci_device_id() const { return info.device_id; }
+
+    /**
+     * @return PCI revision value from sysfs.
+     * TODO: target for removal; upper layers should not care about this.
+     */
+    int get_pci_revision() const { return revision; }
+
+    /**
+     * @return what architecture this device is (e.g. Wormhole, Blackhole, etc.)
+     */
+    tt::ARCH get_arch() const { return arch; }
+
+    // Note: byte_addr is (mostly but not always) offset into BAR0.  This
+    // interface assumes the caller knows what they are doing - but it's unclear
+    // how to use this interface correctly without knowing details of the chip
+    // and its state.
+    // TODO: build a proper abstraction for IO.  At this level, that is access
+    // to registers in BAR0 (although possibly the right abstraction is to add
+    // methods that perform specific operations as opposed to generic register
+    // read/write methods) and access to segments of BAR0/4 that are mapped to
+    // NOC endpoints.  Probably worth waiting for the KMD to start owning the
+    // resource management aspect of these PCIe->NOC mappings (the "TLBs")
+    // before doing too much work here...
```

Comment:
FYI on the redesign.
As I mentioned in another comment, this is arch specific, and will be raised to TTDevice. For the part of "use this interface correctly without knowing details of the chip and its state", yes, that will be hidden probably behind tt::umd::Core or tt::umd::AbstractIO

No action on this comment.